brickpi3/brickpi3_spi.c

```diff
@@ -536,8 +537,12 @@ static int brickpi3_detect(struct brickpi3 *bp)
 
 	dev_info(dev, "FW: %u.%u.%u\n", value / 1000000 % 1000000,
 		 value / 1000 % 1000, value % 1000);
-	if (value < BRICKPI3_MIN_FIRMWARE_VERSION)
-		return -EINVAL;
+///	if (value < BRICKPI3_MIN_FIRMWARE_VERSION)
+	if (value < BRICKPI3_REQUIRED_FIRMWARE_VERSION || value > BRICKPI3_REQUIRED_FIRMWARE_VERSION + 999) {
+	  dev_err(dev, "Unsupported firmware version, expecting %u.%u.%u", BRICKPI3_REQUIRED_FIRMWARE_VERSION / 1000000 % 1000000,
```

Comment:
Also, I would like to replace the last `%u` with `x` so that the messages says `..., expecting 1.2.x` instead of `... 1.2.0`.